lib/raidenclient/RaidenClient.ts

```diff
@@ -201,12 +203,31 @@ class RaidenClient extends SwapClient {
     // not implemented, raiden does not use invoices
   }
 
-  public getRoutes =  async (_amount: number, _destination: string) => {
-    // stub placeholder, query routes not currently implemented in raiden
-    // assume a fixed lock time of 100 Raiden's blocks
-    return [{
+  public getRoutes =  async (amount: number, destination: string, currency: string) => {
+    // a query routes call is not currently provided by raiden
+
+    /** A placeholder route value that assumes a fixed lock time of 100 Raiden's blocks. */
+    const placeholderRoute = {
       getTotalTimeLock: () => 101,
-    }];
+    };
+
+    if (this.directChannelChecks) {
+      // temporary check for a direct channel in raiden
+      const tokenAddress = this.tokenAddresses.get(currency);
+      const channels = await this.getChannels(tokenAddress);
+      for (const channel of channels) {
+        if (channel.partner_address && channel.partner_address === destination) {
+          if (channel.balance >= amount) {
+            return [placeholderRoute];
+          } else {
+            return []; // we have a direct channel but it doesn't have enough balance, return no routes
+          }
+        }
+      }
+      return []; // no direct channels, return no routes
```

Comment:
I definitely would print a `NoDirectChannels` error here, not generic `NoRouteFound`. Maybe sth along the lines. `Swap failed due to no direct raiden channel, please make sure to have a direct raiden channel with this peers and try swapping again.`

Reply:
I can log messages like that here explaining that either there is no direct channel with this peer or that we have one but it doesn't have enough balance, but note that either way this will go in the log output and not in a grpc response where it's front and center for a user who has just issued an `xucli` command can see.

Reply:
I added log statements - I'm hesitant though to introduce a new swap failure reason tho since it would require some non-trivial refactoring and since this is a temporary safeguard that we may remove later. `NoRouteFound` I think is descriptive enough, it tells the user that there's no suitable route to make a payment and the standard corrective action for that in any case is to open a channel to the peer we want to swap with.

Reply:
Alright. We'll make it clear in the documentation.